lib/Dialect/AIEVec/IR/AIEVecOps.cpp

```diff
@@ -213,22 +213,36 @@ OpFoldResult SRSOp::fold(FoldAdaptor adaptor) {
   auto srcDefOp = getSource().getDefiningOp();
   if (!srcDefOp)
     return nullptr;
+
   auto upsOp = dyn_cast<UPSOp>(srcDefOp);
   if (!upsOp)
     return nullptr;
+
+  auto shiftDefOp = getShift().getDefiningOp();
+  if (!shiftDefOp)
+    return nullptr;
+
+  auto constOp = dyn_cast<arith::ConstantOp>(shiftDefOp);
+  if (!constOp)
+    return nullptr;
```

Comment:
This test can be done in one go, like so:
```
auto constOp = getShift().getDefiningOp<arith::ConstantOp>();
```

For future reference :-) @jamestcl-amd @david-vc